docs/developer-env.md

```diff
@@ -0,0 +1,473 @@
+(developer-env)=
+
+# Environment setup
+
+Setting up a Nextflow development environment is a prerequisite for creating, testing, and optimizing data analysis pipelines. The steps below outline recommended tools for setting up an optimal Nextflow development environment.
```

Comment:
A brief TOC here would help I think, not sure if that's non-standard for these docs though.